### ISSUE TITLE
[BUG] Photos with wrong date in timeline

### ISSUE THREAD

**Issue by yotis1982** [immich-app/immich#4992]
### The bug

Thank you checking this and all effort on the project!

The photos in cause are loaded from an external library, they have the correct date (checked with exiftool and even showed on the Info directly in the Immich app on iOS - see photo 1 below):

![IMG_7664](https://github.com/immich-app/immich/assets/1459159/511fb25e-222a-42fe-bcea-98f80b6e5c81)

Howere somehow the mobile app is building the timeline with different dates (see photo 2 below):

![IMG_7665](https://github.com/immich-app/immich/assets/1459159/e8d41888-c5b3-4c49-b85f-16af492dd9ba)

When I checked on web page (from computer or iPhone browser) the timeline is displayed correctly.

If you need more details please let me know.

Thank you!

### The OS that Immich Server is running on

Debian 12

### Version of Immich Server

1.85.0

### Version of Immich Mobile App

1.85.0

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
Not relevant in this case.
```

### Your .env content

```Shell
Not relevant in this case.
```

### Reproduction steps

```bash
1.Close the app
2.Reopen
3.Timeline still wrong

Or

1.Logout
2.Login
3.Timeline still wrong

Or

1.Delete the app
2.Reinstall the app On iOS (iPhone 13, iOS 17.0.3 If relevant)
3.Login
4.Timeline still wrong
```

**Comment by alextran1502** [comment]
Hello, can you share the file for troubleshooting?

**Comment by yotis1982** [comment]
Thanks for the quick reply.

I've attached the photos and here is what I see with exiftool for Data entries.

![DSC_6712](https://github.com/immich-app/immich/assets/1459159/2c5df843-fbca-41b5-94c1-cade6edb964a)

$ exiftool DSC_6712.JPG | grep Date
File Modification Date/Time     : 2023:11:08 14:29:00-08:00
File Access Date/Time           : 2023:11:12 16:04:17-08:00
File Inode Change Date/Time     : 2023:11:08 14:29:00-08:00
Modify Date                     : 2007:07:28 13:39:29
Date/Time Original              : 2007:07:28 13:39:29
Create Date                     : 2007:07:28 13:39:29
Create Date                     : 2007:07:28 13:39:29.10
Date/Time Original              : 2007:07:28 13:39:29.10
Modify Date                     : 2007:07:28 13:39:29.10

**Comment by alextran1502** [comment]
Hello, I cannot reproduce this issue 🤔 I uploaded the image, which shows correctly on both the web and the mobile app on July 28th 2007

**Comment by alextran1502** [comment]
Did you have the microservices container run correctly?

**Comment by yotis1982** [comment]
I thought the same, but seems is working fine. I did open the page on a browser on my iPhone. The browser had no cache or history of the Immich url. It shows fine, the picture is displayed in 2007, timeline is ok.
It seems that is only in the iOS app.

I have an Android phone and will install the app there tomorrow to test and see how the timeline is shown and update here.

Thanks for your efforts to reply and check on this issue!

**Comment by yotis1982** [comment]
I went on and checked the asset for the photo in the external library:


immich=# SELECT * FROM "assets" WHERE "originalFileName" = 'DSC_6712';
```
                  id                  | deviceAssetId |               ownerId                |    deviceId    | type  |                             originalPath                             |                                             resizePath                                             |       fileCreatedAt        |       fileModifiedAt       | isFavorite | duration |                                              webpPath                                              | encodedVideoPath |                  checksum                  | isVisible | livePhotoVideoId |           updatedAt           |           createdAt           | isArchived | originalFileName | sidecarPath | isReadOnly |                  thumbhash                   | isOffline |              libraryId               | isExternal | deletedAt |      localDateTime       | stackParentId
--------------------------------------+---------------+--------------------------------------+----------------+-------+----------------------------------------------------------------------+----------------------------------------------------------------------------------------------------+----------------------------+----------------------------+------------+----------+----------------------------------------------------------------------------------------------------+------------------+--------------------------------------------+-----------+------------------+-------------------------------+-------------------------------+------------+------------------+-------------+------------+----------------------------------------------+-----------+--------------------------------------+------------+-----------+--------------------------+---------------

 55950822-0300-41d1-b985-91c974f87c23 | DSC_6712.JPG  | 90cf567b-e66f-468a-9221-b8ac7d819591 | Library Import | IMAGE | /mnt/media/Photos/Memories/austria/2007.07.28-waterfall/DSC_6712.JPG | upload/thumbs/90cf567b-e66f-468a-9221-b8ac7d819591/55/95/55950822-0300-41d1-b985-91c974f87c23.jpeg | 2023-11-08 22:29:00.141+00 | 2023-11-08 22:29:00.141+00 | f          |          | upload/thumbs/90cf567b-e66f-468a-9221-b8ac7d819591/55/95/55950822-0300-41d1-b985-91c974f87c23.webp |                  | \x3dfeaae1d47ad815dad6a5da5e65b8e1d0df9849 | t         |                  | 2023-11-09 06:58:42.211579+00 | 2023-11-09 01:05:26.970711+00 | f          | DSC_6712         |             | t          | \x55e80915027868db57f67986798b7698c540310767 | f         | bf648e91-c2ad-46df-9b41-07c1b99da241 | t          |           | 2007-07-28 13:39:29.1+00 |
(1 row)
```

and then checked the thumb / webp files in the output of the query:

$ exiftool /mnt/media/immich/upload/thumbs/90cf567b-e66f-468a-9221-b8ac7d819591/55/95/55950822-0300-41d1-b985-91c974f87c23.jpeg | grep Date
File Modification Date/Time     : 2023:11:08 21:40:36-08:00
File Access Date/Time           : 2023:11:12 15:27:20-08:00
File Inode Change Date/Time     : 2023:11:08 21:40:36-08:00
Modify Date                     : 2007:07:28 13:39:29
Date/Time Original              : 2007:07:28 13:39:29
Create Date                     : 2007:07:28 13:39:29
Profile Date Time               : 2004:08:13 12:18:06
Create Date                     : 2007:07:28 13:39:29.10
Date/Time Original              : 2007:07:28 13:39:29.10
Modify Date                     : 2007:07:28 13:39:29.10

exiftool /mnt/media/immich/upload/thumbs/90cf567b-e66f-468a-9221-b8ac7d819591/55/95/55950822-0300-41d1-b985-91c974f87c23.webp | grep Date
File Modification Date/Time     : 2023:11:08 17:14:27-08:00
File Access Date/Time           : 2023:11:12 15:27:09-08:00
File Inode Change Date/Time     : 2023:11:08 17:14:27-08:00
Profile Date Time               : 2004:08:13 12:18:06
Modify Date                     : 2007:07:28 13:39:29
Date/Time Original              : 2007:07:28 13:39:29
Create Date                     : 2007:07:28 13:39:29
Create Date                     : 2007:07:28 13:39:29.10
Date/Time Original              : 2007:07:28 13:39:29.10
Modify Date                     : 2007:07:28 13:39:29.10

All seems fine...

**Comment by alextran1502** [comment]
@shenlong-tanwen When you have a chance, can you help testing upload this photos on your instance and see if it shows up correctly on the mobile app in 2007?

**Comment by yotis1982** [comment]
Here is what I did to resolve the issue.
"Force re-scan all library files" option on my external library. The other two Scan options made no difference.

If I check in the SQL DB (same command like above) nothing changed in the output (at least from what I see comparing the fields), which makes me wonder what actually did change.

I'll try to run another test by adding some new content to my external libraries, then
(1) run "Scan New Library Files" -> check result and I should see the new files with the correct date on the timeline, if not
(2) run "Re-scan All Library Files" -> see if now the new files are displayed correctly.

This will conclude if the above was just some unfortunate incident or there is a problem with new files in External Libraries. 
If we need to run all the time when adding new content the "Force re-scan..." this may get tricky with large external libraries in terms of resource consumption.

I'll update this after my test above.

**Comment by yotis1982** [comment]
I ran the above test and added new content to external libraries, did a "Scan New Library Files", refresh browser and app on mobile -> result is that photos appear correctly in the timeline at their original date.

I will conclude that the above is not a Bug and just an unfortunate incident.

Thank you all for reading these posts and for all the effort to reply and develop this great product.

**Comment by ToddNelson-RPCI** [comment]
Just wanted to report that I've been having the same problem for quite a while, possibly since I first started using Immich in July 2022, but I'm just now getting around to fixing it.  The solution for me was the same, a full re-scan of the entire library.  Only thing I can think of is there was some old bug in the Immich photo scanner years ago that has since been resolved.  By rescanning it's now properly capturing the dates of photos.  I didn't even need to log out/back into the iOS app, it was just working properly upon launch after the rescan.
